File: Java Spring Boot Project/CompanyDirectory/src/main/java/com/mindthecode/CompanyDirectory/models/responses/GenericResponse.java
package com.mindthecode.CompanyDirectory.models.responses;

public class GenericResponse<T> {

    private T data;
    private ErrorResponse error;

    public GenericResponse(ErrorResponse error) {
        this.error = error;
    }

    public GenericResponse(T data) {
        this.data = data;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public ErrorResponse getError() {
        return error;
    }

    public void setError(ErrorResponse error) {
        this.error = error;
    }
}

File: Java Spring Boot Project/company_directory_app/lib/models/responses/generic_response.dart
import 'package:company_directory_app/models/responses/error_response.dart';

class GenericResponse<T> {
  final T data;
  final ErrorResponse error;

  GenericResponse({this.data, this.error});

  factory GenericResponse.fromJson(Map<String, dynamic> json) {
    return GenericResponse(
      data: json['data'],
      error: json['error'],
    );
  }
}
